echopype/echodata/zarr_combine.py

```diff
@@ -0,0 +1,742 @@
+from collections import defaultdict
+from typing import Dict, Hashable, List, Optional, Set, Tuple
+
+import dask
+import dask.array
+import numpy as np
+import pandas as pd
+import xarray as xr
+import zarr
+
+from ..utils.coding import COMPRESSION_SETTINGS
+from ..utils.io import get_zarr_compression
+from ..utils.prov import echopype_prov_attrs
+from .api import open_converted
+from .echodata import EchoData
+
+
+class ZarrCombine:
+    """
+    A class that combines a list of EchoData objects by
+    creating a Zarr store and appending each group's
+    Dataset to the store.
+    """
+
+    def __init__(self):
+
+        # all possible time dimensions
+        self.possible_time_dims = {"time1", "time2", "time3", "ping_time"}
+
+        # all possible dimensions that we will append to (mainly time dims)
+        self.append_dims = {"filenames"}.union(self.possible_time_dims)
+
+        # encodings associated with lazy loaded variables
+        self.lazy_encodings = ["chunks", "preferred_chunks"]
+
+        # defaultdict that holds every group's attributes
+        self.group_attrs = defaultdict(list)
+
+        # The sonar_model for the new combined EchoData object
+        self.sonar_model = None
+
+    def _check_ascending_ds_times(self, ds_list: List[xr.Dataset], ed_name: str) -> None:
+        """
+        A minimal check that the first time value of each Dataset is less than
+        the first time value of the subsequent Dataset. If each first time value
+        is NaT, then this check is skipped.
+
+        Parameters
+        ----------
+        ds_list: List[xr.Dataset]
+            List of Datasets to be combined
+        ed_name: str
+            The name of the ``EchoData`` group being combined
+        """
+
+        # get all time dimensions of the input Datasets
+        ed_time_dim = set(ds_list[0].dims).intersection(self.possible_time_dims)
+
+        for time in ed_time_dim:
+
+            # gather the first time of each Dataset
+            first_times = []
+            for ds in ds_list:
+
+                times = ds[time].values
+                if isinstance(times, np.ndarray):
+                    # store first time if we have an array
+                    first_times.append(times[0])
```

Comment:
I think checking for the first non-NaT times and not just the first element would be more general. Do you see a problem with doing so?

Reply:
I have come across the case where the `Platform` group's `time1` variable only has a single `NaT` value (I think this is manually set by us). In this case, checking for the first non-NaT time would not work. That's really the only issue I can see.

Reply:
Ah I see, ok, then we can just do the first element then.